src/misc/hug.ts

```diff
@@ -0,0 +1,30 @@
+import { PlayerToggleSneakEvent } from 'org.bukkit.event.player';
+import { Particle } from 'org.bukkit';
+import { Player } from 'org.bukkit.entity';
+
+const HUG_COOLDOWN = 1250; // milliseconds
+const hugCooldowns = new Set<Player>();
+
+registerEvent(PlayerToggleSneakEvent, async (event) => {
+    if(!event.isSneaking()) return;
+    if(hugCooldowns.has(event.player)) return;
+    const location = event.player.location.add(event.player.location.direction.multiply(0.7));
+    for(var player of location.getNearbyPlayers(0.5)) {
```

Comment:
```suggestion
    for(let player of location.getNearbyPlayers(0.5)) {
```
Prefer const and let over var